File: Loom.Core/Models/GenerationOptions.vb
Namespace Models
    ''' <summary>
    ''' Technical parameters
    ''' </summary>
    Public Class GenerationOptions
        Public Property Temperature As Single?
        Public Property TopK As Single? ' token filter
        Public Property MaxTokens As Integer? ' token output limits
        Public Property StopSequences As List(Of String)
        Public Property JsonMode As Boolean = False

    End Class
End Namespace

File: implementations/csharp/src/Loom.Core/Models/GenerationOptions.cs
using System.Collections.Generic;

namespace Loom.Core.Models
{
    /// <summary>
    /// Technical parameters
    /// </summary>
    public class GenerationOptions
    {
        public float? Temperature { get; set; }
        public float? TopK { get; set; } // token filter
        public int? MaxTokens { get; set; } // token output limits
        public List<string> StopSequences {  get; set; }
        public bool JsonMode { get; set; } = false;
    }
}
